packages/gatsby/cache-dir/loader.js

```diff
@@ -55,7 +55,7 @@ const loadPageDataJson = loadObj => {
       // If the request was for a 404 page and it doesn't exist, we're done
       if (pagePath === `/404.html`) {
         return Object.assign(loadObj, {
-          status: `failure`,
+          status: `error`,
```

Comment:
this is tricky - if for some reason page-data for 404.html doesn't exist and we set failure then we hit this code path:
https://github.com/gatsbyjs/gatsby/blob/8fd5877ed74d89d17633ea380222d8b9d7a9fc09/packages/gatsby/cache-dir/loader.js#L186-L191

but this also just keep user on current page

I do like throwing error so error tracked could pick this up, but I feel like trying no navigate to page will be better end user experience than keeping them on current page - would still love some error for error trackers tho